tabular/src/autogluon/tabular/models/knn/knn_model.py

```diff
@@ -242,6 +244,11 @@ def sample_func(chunk, frac):
             idx = set(idx)
             self._X_unused_index = [i for i in range(num_rows_max) if i not in idx]
         return self.model
+    
+    def _get_maximum_resources(self) -> Dict[str, float]:
+        return {
+            "num_cpus": min(32, ResourceManager.get_cpu_count())
+        }
```

Comment:
Shouldn't need to use `min` here, can just be `32` right?

Also, change default_resources:

```python3
    def _get_default_resources(self):
        # use at most 32 cpus to avoid OpenBLAS error: https://github.com/autogluon/autogluon/issues/1020
        num_cpus = min(32, ResourceManager.get_cpu_count())
        num_gpus = 0
        return num_cpus, num_gpus
```

to

```python3
    def _get_default_resources(self):
        num_cpus = ResourceManager.get_cpu_count()
        num_gpus = 0
        return num_cpus, num_gpus
```

Then, move the ` # use at most 32 cpus to avoid OpenBLAS error: https://github.com/autogluon/autogluon/issues/1020` to the maximum resources method